### PR TITLE
WIP/POC: Remove exclusive range syntax

### PR DESCRIPTION
**Pull request by vegai**
Expendable proof of concept for removing exclusive ranges (...) from the language.

Discussed in https://github.com/crystal-lang/crystal/issues/4214

### CODE REVIEW

File: spec/std/array_spec.cr [hunk 1]
```diff
@@ -273,7 +273,7 @@ describe "Array" do
       a.should eq([1, 2, 6, 5])
 
       a = [1, 2, 3, 4, 5]
-      a[1...1] = 6
+      a[1..0] = 6
```
Comment (bew) [inline spec/std/array_spec.cr]: Are you about the `1..0` here?

File: spec/std/range_spec.cr [hunk 2]
```diff
@@ -291,16 +225,12 @@ describe "Range" do
     end
 
     it "is empty with ... and begin > end" do
```
Comment (bew) [inline spec/std/range_spec.cr]: You should simply remove the test

Comment (bew) [inline spec/std/range_spec.cr]: change the test name

Comment (bew) [inline spec/std/range_spec.cr]: ...  In test name

Comment (bew) [inline spec/std/range_spec.cr]: ... In test name

Comment (bew) [inline spec/std/range_spec.cr]: ... In test name

Comment (bew) [inline spec/std/range_spec.cr]: ... In test name

File: spec/std/range_spec.cr [hunk 3]
```diff
@@ -316,44 +246,17 @@ describe "Range" do
...
-
-      iter.rewind
-      iter.next.should eq(1)
-    end
-
     it "is empty with .. and begin > end" do
       (1..0).step(1).to_a.empty?.should be_true
     end
 
     it "is empty with ... and begin > end" do
```
Comment (bew) [inline spec/std/range_spec.cr]: ... In test name

**Comment by vegai** [comment]
Closing to stop spamming the PR queue with this WIP.

**Pull request closed without merging**